test.cpp

```diff
@@ -1 +1,35 @@
+#include "gtest/gtest.h"
+#include "coordinates.h"
+#include <cmath>
+#include <string>
 
+int main(int argc, char **argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
+// tests go here 
+TEST(Haversine, usingCSbuilding) { 
```

Comment:
Using testing is a good way to avoid errors, your TEST method is a good way to do this!